drivers/periph_common/Kconfig.wdt

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+menuconfig MOD_PERIPH_WDT
+    bool "Watchdog Timer peripheral driver"
+    depends on HAS_PERIPH_WDT
+    select MOD_PERIPH_COMMON
+
+if MOD_PERIPH_WDT
+
+if TEST_KCONFIG
+
+config MOD_PERIPH_INIT_WDT
+    bool "Auto initialize the Watchdog Timer peripheral"
+    default y if MOD_PERIPH_INIT
+    depends on MOD_PERIPH_WDT
+
+endif # TEST_KCONFIG
+
+config WDT_WARNING_PERIOD
```

Comment:
This option should be visible during migration. That's why in the current file the symbol `KCONFIG_MODULE_PERIPH_WDT` is defined, which depends on `MODULE_PERIPH_WDT`. After migration of course this will look just like you modelled it, but for now we need both symbols.

I mean something like
```kconfig
# This is the migration part
if TEST_KCONFIG

menuconfig MOD_PERIPH_WDT
    bool "Watchdog Timer peripheral driver"
    depends on HAS_PERIPH_WDT
    select MOD_PERIPH_COMMON

config MOD_PERIPH_INIT_WDT
    bool "Auto initialize the Watchdog Timer peripheral"
    default y if MOD_PERIPH_INIT
    depends on MOD_PERIPH_WDT

endif # TEST_KCONFIG

# This is what can be configured now
menuconfig KCONFIG_MODULE_PERIPH_WDT
    bool "Watchdog Timer peripheral driver"
    depends on MODULE_PERIPH_WDT

if KCONFIG_MODULE_PERIPH_WDT

config WDT_WARNING_PERIOD
    int "Warning period (in ms)"
    depends on HAS_PERIPH_WDT_WARNING_PERIOD
    help
        Period in ms before reboot where wdt_cb() is executed.

# Include CPU specific configurations
osource "$(RIOTCPU)/$(CPU)/periph/Kconfig.wdt"

endif # KCONFIG_MODULE_PERIPH_WDT

config HAS_PERIPH_WDT_WARNING_PERIOD
    bool
    help
        Indicates that a CPU provides a warning period configuration option.
```

Reply:
So right now you either get modules or configuration?

Reply:
Well, modules are not meant to be used during migration besides for testing the dependencies, so yes. The user should the the configurations (of the modules in `USEMODULE`) as usual.

Reply:
What I mean is that whith this approach I can either see configurations options or test dependencies, but not both at the same time, is that the approach we are taking during migration? Asking just to be sure of the expect output.

Reply:
Yes, that's correct. After modelling all the modules with the dependencies the `menuconfig`s with the `KCONFIG_MODULE_` symbols are removed and replaced by the proper `MOD_` (soon to be `MODULE_`) symbols.